web/state/RoleManager.js

```diff
@@ -115,6 +116,18 @@ class RoleManager {
       currentRoleData[roleLabel.toLowerCase()].user = user;
     }
 
+    if (SYNC_BOT_ID){
```

Comment:
Do you mind extracting this to a dedicated function to make this function more readable?